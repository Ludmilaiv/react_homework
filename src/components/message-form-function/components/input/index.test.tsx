import { Input } from './';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Input', () => {
  it('render component', () => {
    render(<Input className='test' placeholder='text' value=''/>);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Input className='test' placeholder='text' value=''/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Input className='test' placeholder='text' value='my text'/>);
    expect(screen.getByDisplayValue(/my text/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(<>
      <Input className='test' placeholder='text' value='text'/>
      <Input className='test' placeholder='text' value='text'/>
      <Input className='test' placeholder='text' value='text'/>
    </>
    );

    expect(screen.queryAllByDisplayValue('text').length).toBe(3);
  });

  it('input change with fireEvent', () => {
    const mockHandler = jest.fn();
    render(<Input className='test' onChange={mockHandler} placeholder='text' value=''/>);
    fireEvent.change(screen.getByPlaceholderText('text'));
    setTimeout(() => expect(mockHandler).toHaveBeenCalledTimes(1), 100);
  });

  it('input async change', async () => {
    const mockHandler = jest.fn();
    render(<Input className='test' onChange={() => setTimeout(mockHandler, 2000)} placeholder='text' value=''/>);
    fireEvent.change(screen.getByPlaceholderText('text'));
    setTimeout(() => expect(mockHandler).toHaveBeenCalledTimes(1), 2100);
  });
});