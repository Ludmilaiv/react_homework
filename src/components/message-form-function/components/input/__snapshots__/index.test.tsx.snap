// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with snapshot 1`] = `
<DocumentFragment>
  <input
    class="test"
    placeholder="text"
    type="text"
    value=""
  />
</DocumentFragment>
`;
